From d8372ba8ce288acdfce67cb873b2a741785c2e88 Mon Sep 17 00:00:00 2001
From: NeilBrown <neil@brown.name>
Date: Sat, 29 Dec 2018 13:58:09 +1100
Subject: [PATCH] lib: don't depend on linux headers being installed.
Git-commit: d8372ba8ce288acdfce67cb873b2a741785c2e88
Patch-mainline: v4.20
References: bsc#1130972

gen_crc64table requires linux include files to be installed in
/usr/include/linux.  This is a new requrement so hosts that could
previously build the kernel, now cannot.

gen_crc64table makes this requirement by including <linux/swab.h>, but
nothing from that header is actaully used.

So remove the #include, so that the linux headers no longer need to be
installed.

Fixes: feba04fd2cf8 ("lib: add crc64 calculation routines")
Signed-off-by: NeilBrown <neil@brown.name>
Acked-by: Coly Li <colyli@suse.de>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 lib/gen_crc64table.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/gen_crc64table.c b/lib/gen_crc64table.c
index 9011926e4162..094b43aef8db 100644
--- a/lib/gen_crc64table.c
+++ b/lib/gen_crc64table.c
@@ -16,8 +16,6 @@
 #include <inttypes.h>
 #include <stdio.h>
 
-#include <linux/swab.h>
-
 #define CRC64_ECMA182_POLY 0x42F0E1EBA9EA3693ULL
 
 static uint64_t crc64_table[256] = {0};
-- 
2.16.4

